modules/dag-l0/src/main/scala/io/constellationnetwork/dag/l0/http/routes/DelegatedStakesRoutes.scala

```diff
@@ -0,0 +1,121 @@
+import io.constellationnetwork.node.shared.domain.snapshot.storage.SnapshotStorage
+import io.constellationnetwork.node.shared.infrastructure.snapshot.storage.SnapshotLocalFileSystemStorage
+import io.constellationnetwork.routes.internal._
+import io.constellationnetwork.schema._
+import io.constellationnetwork.schema.address.Address
+import io.constellationnetwork.schema.delegatedStake.UpdateDelegatedStake
+import io.constellationnetwork.schema.peer.PeerId
+import io.constellationnetwork.security.signature.Signed
+
+import derevo.cats.{eqv, show}
+import derevo.circe.magnolia.encoder
+import derevo.derive
+import eu.timepit.refined.auto._
+import org.http4s.HttpRoutes
+import org.http4s.circe.CirceEntityCodec.circeEntityEncoder
+import org.http4s.dsl.Http4sDsl
+import org.typelevel.log4cats.slf4j.Slf4jLogger
+
+final case class DelegatedStakesRoutes[F[_]: Async](
+  mkCell: DelegatedStakeOutput => Cell[F, StackF, _, Either[CellError, Ω], _],
+  validator: UpdateDelegatedStakeValidator[F],
+  snapshotStorage: SnapshotStorage[F, GlobalIncrementalSnapshot, GlobalSnapshotInfo],
+  fullGlobalSnapshotStorage: SnapshotLocalFileSystemStorage[F, GlobalSnapshot],
+  nodeStorage: NodeStorage[F]
+) extends Http4sDsl[F]
+    with PublicRoutes[F] {
+
+  import org.http4s.circe.CirceEntityCodec.circeEntityDecoder
+
+  private val logger = Slf4jLogger.getLoggerFromName[F]("DelegatedStakesLogger")
+
+  protected val prefixPath: InternalUrlPrefix = "/"
+
+  private def readLatestFullSnapshot(maybeOrdinal: Option[SnapshotOrdinal]): F[Option[Signed[GlobalSnapshot]]] =
+    maybeOrdinal.traverse(fullGlobalSnapshotStorage.read).map(_.flatten)
+
+  private def getLatestFullSnapshot: F[Option[Signed[GlobalSnapshot]]] =
+    for {
+      maybeOrdinal <- snapshotStorage.headSnapshot.map(_.map(_.ordinal))
+      maybeSnapshot <- readLatestFullSnapshot(maybeOrdinal)
+    } yield maybeSnapshot
+
+  @derive(eqv, show, encoder)
+  case class DelegatedStakesInfo(
+    address: Address,
+    activeDelegatedStakes: List[Signed[UpdateDelegatedStake.Create]],
+    pendingWithdrawals: List[(Signed[UpdateDelegatedStake.Withdraw], SnapshotOrdinal)]
+  )
+
+  protected val public: HttpRoutes[F] = HttpRoutes.of[F] {
+    case req @ POST -> Root / "delegated-stakes" =>
+      getLatestFullSnapshot.flatMap {
+        case Some(lastSnapshot) =>
+          req
+            .as[Signed[UpdateDelegatedStake.Create]]
+            .flatMap(signed => validator.validateCreateDelegatedStake(signed, lastSnapshot.info, lastSnapshot.epochProgress))
+            .flatTap {
+              case Valid(signed) =>
+                logger.info(s"Accepted create delegated stake from ${signed.proofs.map(_.id).map(PeerId.fromId(_))}")
+              case Invalid(errors) =>
+                logger.warn(s"Invalid create delegated stake: $errors")
+            }
+            .flatMap {
+              case Valid(signed)   => mkCell(CreateDelegatedStakeOutput(signed)).run()
```

Comment:
Nitpick: this is potentially out of scope of the PR, but anywhere you can bypass this mkCell and just directly enqueue, all the better. We're not using the rest of that functionality anyways